modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/GlassViewEventHandler.java

```diff
@@ -672,6 +673,16 @@ public double[] getInputMethodCandidatePos(int offset) {
         double[] ret = new double[2];
         ret[0] = p2d.getX();
         ret[1] = p2d.getY();
+
+        for (Screen scr : Screen.getScreens()) {
+            Rectangle2D bounds = new Rectangle2D(scr.getX(), scr.getY(), scr.getWidth(), scr.getHeight());
```

Comment:
To correctly handle the case of a window straddling two screens, you need the screen associated with the window, not the screen that the anchor point intersects.